jax/experimental/compilation_cache/compilation_cache.py

```diff
@@ -145,6 +154,40 @@ def _hash_platform(hash_obj, backend):
   _hash_string(hash_obj, backend.platform_version)
   _hash_string(hash_obj, backend.runtime_type)
 
+_xla_flags_to_exclude_from_cache_key = [
+    "--xla_dump_compress_protos",
+    "--xla_dump_module_metadata",
+    "--xla_dump_max_hlo_modules",
+    "--xla_dump_include_timestamp",
+    "--xla_dump_hlo_pass_re",
+    "--xla_dump_hlo_module_re",
+    "--xla_dump_hlo_snapshots",
+    "--xla_dump_fusion_visualization",
+    "--xla_dump_hlo_as_url",
+    "--xla_dump_hlo_as_proto",
+    "--xla_dump_hlo_as_text",
+    "--xla_dump_to",
+    "--xla_force_host_platform_device_count",
+    "--xla_dump_disable_metadata",
+    "--xla_dump_hlo_pipeline_re",
+]
+
+def _hash_xla_flags(hash_obj):
+  # All XLA flags must be specified via the XLA_FLAGS env var, both inside and
+  # outside of google (I think).
```

Comment:
While this may be true externally, it's certainly not true internally, especially for TPUs. I'll send you an email internally with some references.

Reply:
I added logic to also look at sys.argv.

Reply:
LGTM, but add explicit note wrt XLA absl flags must be passed in the form of `--flag_name=flag_value`, not with spaces?
(And confirmatory action sent internally.)